### PR TITLE
Add support for proxy.

### PR DESCRIPTION
**Pull request by mohsinenar**

### CODE REVIEW

File: tests/zap_agent_test.py [hunk 1]
```diff
@@ -172,3 +171,31 @@ def testUseVpn_whenConfigFile_callVPN(
 
         assert subprocess_mocker.call_count == 1
         assert subprocess_mocker.call_args_list[0].args[0] == ["wg-quick", "up", "wg0"]
+
+
+def testAgentZap_whenProxyIsProvided_RunScanWithProxyArguments(
+    scan_message_2, test_agent_with_proxy, mocker, agent_mock
+):
```
Comment (rabsondev) [inline tests/zap_agent_test.py:178]: Typing.

File: tests/conftest.py [hunk 2]
```diff
@@ -87,6 +89,27 @@ def test_agent_with_url_scope():
         return zap_agent.ZapAgent(definition, settings)
 
 
+@pytest.fixture
+def test_agent_with_proxy():
```
Comment (rabsondev) [inline tests/conftest.py:93]:
```suggestion
def test_agent_with_proxy() -> zap_agent.ZapAgent:
```

File: tests/zap_wrapper_test.py [hunk 3]
```diff
@@ -26,3 +27,31 @@ def testZapWrapperScan_withTimeoutException_raisesValueError(
         zap = zap_wrapper.ZapWrapper(scan_profile="baseline")
         zap.scan(target="https://dummy.com")
     assert run_mock.call_count == 5
+
+
+def testZapWrapperScan_whenProxyNoSchema_shouldNotCallWithProxy(
+    mocker: plugin.MockerFixture,
+):
```
Comment (rabsondev) [inline tests/zap_wrapper_test.py:34]: Typing.

File: agent/zap_wrapper.py [hunk 4]
```diff
@@ -21,11 +22,27 @@
...
 JAVA_COMMAND_TIMEOUT = datetime.timedelta(minutes=60)
 
+ProxyTuple = NamedTuple("ProxyTuple", [("proxy_host", str), ("proxy_port", str)])
+
+
+def _parse_proxy(proxy: str) -> ProxyTuple | None:
+    """Get proxy arguments."""
+    parsed_url = parse.urlparse(proxy)
+    if parsed_url.port is not None and parsed_url.hostname is not None:
+        return ProxyTuple(parsed_url.hostname, str(parsed_url.port))
```
[the remark this answers is not in the document]
Comment (mohsinenar) [reply agent/zap_wrapper.py:32]: Not needed!

File: agent/zap_wrapper.py [hunk 5]
```diff
@@ -5,7 +5,8 @@
 import pathlib
 import subprocess
 import tempfile
-from typing import List, Dict
+from typing import List, Dict, NamedTuple
```
Comment (3asm) [inline agent/zap_wrapper.py:8]: Same here.

File: ostorlab.yaml [hunk 6]
```diff
@@ -104,3 +104,6 @@ args:
     type: "number"
     description: "Max crawl duration in minutes."
     value: 10
+  - name: "proxy"
+    type: "string"
+    description: "Proxy to use for the scan with nuclei."
```
Comment (3asm) [inline ostorlab.yaml:109]:
```suggestion
    description: "Proxy to use for the scan with Zap."
```

File: agent/zap_wrapper.py [hunk 7]
```diff
@@ -5,7 +5,8 @@
 import pathlib
 import subprocess
 import tempfile
-from typing import List, Dict
+from typing import NamedTuple
```
Comment (elyousfi5) [inline agent/zap_wrapper.py:8]:
```suggestion
import typing
```

Comment (mohsinenar) [reply agent/zap_wrapper.py:8]: For `typing` we do import directly

**Review by 3asm — APPROVED** [review]

**Pull request merged**